### ISSUE TITLE
hw02 ready for grading

### ISSUE THREAD

**Issue by hsmohammed** [hsmohammed/STAT545-hw-Mohammed-Hossameldin#2]
Hello, My Homework 02 is ready to be reviewed from the following link:
https://github.com/hsmohammed/STAT545-hw-Mohammed-Hossameldin/tree/master/HW02
SHA:
21dd9c5b4bffd6efab53abdc1f172a5bb075f520

@vincenzocoia @gvdr @ksedivyhaley @JoeyBernhardt @mynamedaike @pgonzaleze @derekcho
Thank you
Hossam

**Comment by Jennica416** [comment]
Hi @hsmohammed ,

Your assignment is great! I appreciated the README file that made navigation really easy.  

Grade: +

Smell test: I thought the functions you used overall in this section were good to answer the questions. I especially like that you used str() to get type and class for all variables in one line of code. Clever.  My own suggestion would be to include short explanations  after the code to help interpret the code (e.g. there are 6 variables after the ncol(gapminder) function.

 Individual variables: Great use of functions and graph type. I would suggest using summary() with the continuous variable as you get additional information (e.g. mean, median) that can help you understand the data. 

Plots: I think this was really well done and the code was good. Again, explanations after the graph would have been appreciated. There is a typo in your code that filters Canada and Mexico - need to capitalize the m (so it would be filter(countries %in% c("Canada", "Mexico")) as the current code won't include Mexico rows 

Other things: Well done on the fancy table, I appreciated being able to look at both the Rmd and md file and thought your reflections on your process were informative. 

Well done,
Jennica

**Comment by ksedivyhaley** [comment]
**Smell test of data:** Partial, see comments
**Explores at least one categorical/quantitative variable:** Yes
**Uses various ggplot types:** Yes (scatter, density, freqpoly, boxplot)
**Uses `filter()`, `select()`, and `%>%`:** Partial (no `select`)
**Bonus (evaluate code, tables, more dplyr):** Attempt at kable.
**Reflection on process:** Yes

**Comments:**

- Your "smell test" should be more explicit about what information you're getting from the `str()` and `summary()` functions. Is `str()` giving you the type of the dataset? The type of the variables? What does `summary()` tell you about the size and extent of the data?
- You don't have to extract variables from the gapminder dataset, you don't need to save the results of your functions, and you don't need to call `print`. Just use eg `table(gapminder$continent)`
- Neat job defining the `getmode()` function! A little hard to follow, though. Good place for code comments.
- Try adding more features to your graphs, such as colour or faceting.
-  Try using the `format` argument to fix your kable display.

Your mark will be distributed later. If you would like more feedback, please feel free to message me on slack.
